### PR TITLE
Online state

### PR DESCRIPTION
**Pull request by daniel-lerch**
This pull request implements the protocol changes from skynet-im/skynet#1. Online states are now supported by the server. Other improvements affect FCM which should have less problems with ghost clients now and got some new configuration options. The delivery service no longer sends clients dependencies of other accounts.

### Breaking changes (Client)
- New packets have to be implemented
- Multiple dependencies per account and message have to be supported
- `0x19 KeypairReference` is no longer injected

### Breaking changes (Server)
- Database model changed
- Database reset required due to incompatible channel message updates
- New configuration section `FcmOptions`

### CODE REVIEW

File: src/SkynetServer/Network/Model/ChannelAction.cs [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+using System.Collections.Generic;
+using System.Text;
+
+namespace SkynetServer.Network.Model
+{
+    internal enum ChannelAction
+    {
+        None,
+        Typing,
+        RecordingAudio
```
Comment (Twometer) [inline src/SkynetServer/Network/Model/ChannelAction.cs]: Should we add something like "taking a picture" here? It annoys me when I ask someone on WhatsApp to send an image of something, then they don't read it and i don't know if they are just in the CameraActivity or ignoring me.

Comment (Twometer) [inline src/SkynetServer/Network/Model/ChannelAction.cs]: Should we add something like `TakingPicture` here? I find it annoying when I ask someone on WhatsApp to send an image of something, and then they don't read my messages and I don't know whether they are in the `CameraActivity` or just plain ignoring me.

Comment (daniel-lerch) [reply src/SkynetServer/Network/Model/ChannelAction.cs]: When I'm supposed to take and send multiple pictures to anyone, I usually use the Camera app and share these pictures afterwards. If we want to have something reliable we would have to implement a feature to share your foreground app like Discord does for games. But then you should open an issue in skynet-im/skynet.

Comment (Twometer) [reply src/SkynetServer/Network/Model/ChannelAction.cs]: alright

**Pull request merged**